server/src/main/java/org/elasticsearch/script/AbstractFieldScript.java

```diff
@@ -75,6 +77,7 @@ public AbstractFieldScript(String fieldName, Map<String, Object> params, SearchL
 
         this.fieldName = fieldName;
         Map<String, Object> docAsMap = docAsMap();
+        this.sourcePaths = searchLookup.sourcePaths(fieldName);
```

Comment:
I'm happy to make this change. I do think this will be extended to all available runtime fields, though. Maybe it would make sense to provide an additional constructor or a boolean value to have this be null?

Reply:
I have an instinct that if you implemented it with copy and paste for two field types, like, `keyword` and `ip` or something, and then tried to remove the duplications *something* would pop out. I don't know what. I just think it might be worth trying before adding this for all runtime fields.